src/sockit.cc

```diff
@@ -138,6 +141,31 @@ Sockit::Connect(const Arguments& aArgs) {
     );
   }
 
+  Sockit *sockit = ObjectWrap::Unwrap<Sockit>(aArgs.This());
+
+  // Ensure we're not already attempting to connect.
+  if(sockit->mIsConnecting) {
```

Comment:
It was getting called multiple times which shouldn't be something we allow as it's not allowed in the bsd socket API either.